gdal/frmts/tiledb/tiledbdataset.cpp

```diff
@@ -1376,6 +1460,76 @@ TileDBDataset* TileDBDataset::CreateLL( const char *pszFilename,
         
         poDS->m_schema->set_domain( domain ).set_order({{TILEDB_ROW_MAJOR, TILEDB_ROW_MAJOR}});;
 
+        // register additional attributes to the pixel value, these will be
+        // be reported as subdatasets on future reads
+        poDS->papszAttributes = CSLFetchNameValueMultiple(
+                                                    papszOptions, "TILEDB_ATTRIBUTE" );
+
+        for( int i = 0;
+            poDS->papszAttributes != nullptr &&
+            poDS->papszAttributes[i] != nullptr;
+            i++ )
+        {
+            // modeling additional attributes as subdatasets
+            poDS->bHasSubDatasets = true;
+            // check each attribute is a GDAL source
+            std::unique_ptr<GDALDataset> poAttrDS(
+                (GDALDataset*) GDALOpen( poDS->papszAttributes[i],
+                GA_ReadOnly ));
+
+            if ( poAttrDS != nullptr )
+            {
+                const char* pszAttrName = CPLGetBasename(
+                                            poAttrDS->GetDescription());
+                // check each is co-registered
+                if ( poAttrDS->GetSpatialRef() == poDS->GetSpatialRef() )
```

Comment:
pointer equality testing will not work.  You should use poSRS1->IsSame(poSRS2) (after null-test checking of both side), or accepting both as nullptr

Reply:
noted, thanks, didn't realize that. I am going to relax that check, it is sufficient that the width, height and number of bands is equal for an additional attribute.